### DISCUSSION TITLE
calculate camerarig

### DISCUSSION THREAD

**Discussion by markxma1** [alicevision/Meshroom#1382] [Q&A] [answered]
Hi, I try to calculate camera rig, but with no susses. I cannot find some description how to start it.
I have no exif in images (it was made with opencv).
I have made the folder structure like it was on wikisite:
V1:
```
+ rig/  
|-+ 0/ 
|---- 1.png
|---- 2.png
|---- 3.png
|---- 4.png
|-+ 1/
|---- 1.png
|---- 2.png
|---- 3.png
|---- 4.png
```
and tried to:
V2
```
+ rig/  
|-+ 0/ 
|---- 101.png
|---- 102.png
|---- 103.png
|---- 104.png
|-+ 1/
|---- 201.png
|---- 202.png
|---- 203.png
|---- 204.png
```
 I have made sfm file with  StructurFromMotion (and changed path for V1):

```
{
    "version": [
        "1",
        "0",
        "0"
    ],
    "featuresFolders": [
        "..\\..\\FeatureExtraction\\54f7cb26c30018d5c72e877dd2e64093752516a7"
    ],
    "matchesFolders": [
        "..\\..\\FeatureMatching\\1c80b40d70e5d3cd7eb7aa95df1732cf0d2c60c2"
    ],
    "views": [
        {
            "viewId": "101",
            "poseId": "101",
            "rigId": "3377974952",
            "subPoseId": "0",
            "frameId": "1",
            "intrinsicId": "1",
            "resectionId": "3",
            "path": "C:\/Users\/Xma1\/Desktop\/RIGCHECK\/rig\/0\/1_01.png",
            "width": "1152",
            "height": "1024",
            "metadata": {
                "AliceVision:useWhiteBalance": "1"
            }
        },
        {
            "viewId": "102",
            "poseId": "102",
            "rigId": "3377974952",
            "subPoseId": "0",
            "frameId": "1",
            "intrinsicId": "1",
            "resectionId": "4",
            "path": "C:\/Users\/Xma1\/Desktop\/RIGCHECK\/rig\/0\/1_02.png",
            "width": "1152",
            "height": "1024",
            "metadata": {
                "AliceVision:useWhiteBalance": "1"
            }
        },
        {
            "viewId": "103",
            "poseId": "103",
            "rigId": "3377974952",
            "subPoseId": "0",
            "frameId": "1",
            "intrinsicId": "1",
            "resectionId": "5",
            "path": "C:\/Users\/Xma1\/Desktop\/RIGCHECK\/rig\/0\/1_03.png",
            "width": "1152",
            "height": "1024",
            "metadata": {
                "AliceVision:useWhiteBalance": "1"
            }
        },
        {
            "viewId": "104",
            "poseId": "104",
            "rigId": "3377974952",
            "subPoseId": "0",
            "frameId": "1",
            "intrinsicId": "1",
            "resectionId": "2",
            "path": "C:\/Users\/Xma1\/Desktop\/RIGCHECK\/rig\/0\/1_04.png",
            "width": "1152",
            "height": "1024",
            "metadata": {
                "AliceVision:useWhiteBalance": "1"
            }
        },
        {
            "viewId": "201",
            "poseId": "201",
            "rigId": "3377974952",
            "subPoseId": "1",
            "frameId": "2",
            "intrinsicId": "2",
            "resectionId": "1",
            "path": "C:\/Users\/Xma1\/Desktop\/RIGCHECK\/rig\/1\/2_01.png",
            "width": "1136",
            "height": "1024",
            "metadata": {
                "AliceVision:useWhiteBalance": "1"
            }
        },
        {
            "viewId": "202",
            "poseId": "202",
            "rigId": "3377974952",
            "subPoseId": "1",
            "frameId": "2",
            "intrinsicId": "2",
            "resectionId": "0",
            "path": "C:\/Users\/Xma1\/Desktop\/RIGCHECK\/rig\/1\/2_02.png",
            "width": "1136",
            "height": "1024",
            "metadata": {
                "AliceVision:useWhiteBalance": "1"
            }
        },
        {
            "viewId": "203",
            "poseId": "203",
            "rigId": "3377974952",
            "subPoseId": "1",
            "frameId": "2",
            "intrinsicId": "2",
            "path": "C:\/Users\/Xma1\/Desktop\/RIGCHECK\/rig\/1\/2_03.png",
            "width": "1136",
            "height": "1024",
            "metadata": {
                "AliceVision:useWhiteBalance": "1"
            }
        },
        {
            "viewId": "204",
            "poseId": "204",
            "rigId": "3377974952",
            "subPoseId": "1",
            "frameId": "2",
            "intrinsicId": "2",
            "path": "C:\/Users\/Xma1\/Desktop\/RIGCHECK\/rig\/1\/2_04.png",
            "width": "1136",
            "height": "1024",
            "metadata": {
                "AliceVision:useWhiteBalance": "1"
            }
        }
    ],
    "intrinsics": [
        {
            "intrinsicId": "1",
            "width": "1152",
            "height": "1024",
            "sensorWidth": "-1",
            "sensorHeight": "-0.88888888888888884",
            "serialNumber": "C:\/Users\/Xma1\/Desktop\/RIGCHECK\/rig\/1_rig_3377974952_1",
            "type": "radial3",
            "initializationMode": "unknown",
            "pxInitialFocalLength": "-1",
            "pxFocalLength": "1273.0661519396799",
            "principalPoint": [
                "592.63057643872276",
                "520.61163269857411"
            ],
            "distortionParams": [
                "-0.27809300868050602",
                "0.21421448251830408",
                "-0.13306897839866941"
            ],
            "locked": "0"
        },
        {
            "intrinsicId": "2",
            "width": "1136",
            "height": "1024",
            "sensorWidth": "-1",
            "sensorHeight": "-0.90140845070422537",
            "serialNumber": "C:\/Users\/Xma1\/Desktop\/RIGCHECK\/rig\/1_rig_3377974952_1",
            "type": "radial3",
            "initializationMode": "unknown",
            "pxInitialFocalLength": "-1",
            "pxFocalLength": "1281.7278194445582",
            "principalPoint": [
                "624.80000000000007",
                "539.21837990285076"
            ],
            "distortionParams": [
                "-0.27563193209662706",
                "0.20189655669526185",
                "-0.13472803688025536"
            ],
            "locked": "0"
        }
    ],
    "poses": [
        {
            "poseId": "101",
            "pose": {
                "transform": {
                    "rotation": [
                        "-0.95207427995337945",
                        "0.29792483749536852",
                        "0.069248513735764905",
                        "-0.27443041575685251",
                        "-0.73206697543569232",
                        "-0.62351093846375993",
                        "-0.13506484501444194",
                        "-0.61263262619599013",
                        "0.77874177553373225"
                    ],
                    "center": [
                        "-0.43887460719311427",
                        "2.6541520962647982",
                        "1.1223145980146467"
                    ]
                },
                "locked": "0"
            }
        },
        {
            "poseId": "102",
            "pose": {
                "transform": {
                    "rotation": [
                        "-0.98712361317717823",
                        "0.15991060047569394",
                        "-0.0039461580727899077",
                        "-0.14840460176134274",
                        "-0.92474577562408344",
                        "-0.35045873457722448",
                        "-0.059691259695953153",
                        "-0.34536046432809181",
                        "0.93656986028539002"
                    ],
                    "center": [
                        "-0.15425931866864884",
                        "2.3356783602195952",
                        "0.54395938934500143"
                    ]
                },
                "locked": "0"
            }
        },
        {
            "poseId": "103",
            "pose": {
                "transform": {
                    "rotation": [
                        "-0.99946293762389238",
                        "-0.0085380433584399239",
                        "-0.031637606291069612",
                        "0.0090649862947668556",
                        "-0.99982194994491835",
                        "-0.01654975624644962",
                        "-0.031490670677123037",
                        "-0.016827662462463675",
                        "0.99936238043882508"
                    ],
                    "center": [
                        "0.041542225757798867",
                        "1.8666654981086519",
                        "0.13973066751214838"
                    ]
                },
                "locked": "0"
            }
        },
        {
            "poseId": "104",
            "pose": {
                "transform": {
                    "rotation": [
                        "-0.99550051552495811",
                        "-0.091493800020819332",
                        "-0.024649708868317653",
                        "0.087076386926807664",
                        "-0.98591209097195143",
                        "0.14281124505824941",
                        "-0.037368789508296606",
                        "0.14002226049119762",
                        "0.98944294435688418"
                    ],
                    "center": [
                        "0.15760880451711801",
                        "1.0493955025167438",
                        "-0.12943757956144394"
                    ]
                },
                "locked": "0"
            }
        },
        {
            "poseId": "201",
            "pose": {
                "transform": {
                    "rotation": [
                        "0.94608622537984244",
                        "-0.306946029071687",
                        "0.10346491861324274",
                        "0.30814672809082566",
                        "0.75443430509684906",
                        "-0.57954678263273374",
                        "0.099832099614505967",
                        "0.58018360415488945",
                        "0.80834431856505429"
                    ],
                    "center": [
                        "0.13622154342411341",
                        "1.2590095863385",
                        "-0.0074893227611422877"
                    ]
                },
                "locked": "0"
            }
        },
        {
            "poseId": "202",
            "pose": {
                "transform": {
                    "rotation": [
                        "0.98570058479185707",
                        "-0.16430378679098612",
                        "0.037398165558395716",
                        "0.1678260757363253",
                        "0.93731701528033085",
                        "-0.30540337452116872",
                        "0.015124994014418665",
                        "0.30731267222832548",
                        "0.95148838986292916"
                    ],
                    "center": [
                        "0.15768101194924955",
                        "0.5809058305728817",
                        "-0.076724050724225307"
                    ]
                },
                "locked": "0"
            }
        },
        {
            "poseId": "203",
            "pose": {
                "transform": {
                    "rotation": [
                        "0.99981203074483793",
                        "0.0076029513143481432",
                        "0.017835310740068808",
                        "-0.0080155339201578787",
                        "0.99969925215768241",
                        "0.023176635893617001",
                        "-0.017653735974517134",
                        "-0.023315238936844486",
                        "0.99957228114802177"
                    ],
                    "center": [
                        "0.030411927887684626",
                        "-0.020730414887354796",
                        "0.14239140967987624"
                    ]
                },
                "locked": "0"
            }
        },
        {
            "poseId": "204",
            "pose": {
                "transform": {
                    "rotation": [
                        "0.99578976249783868",
                        "0.087389394282787733",
                        "0.0276738626032864",
                        "-0.090946235539033957",
                        "0.97963868483807304",
                        "0.17898834992817025",
                        "-0.011468702881174559",
                        "-0.18075160009144603",
                        "0.98346190974465564"
                    ],
                    "center": [
                        "-0.0068331324960142908",
                        "-0.8095588586154342",
                        "0.17013899162621515"
                    ]
                },
                "locked": "0"
            }
        }
    ],
    "rigs": [
        {
            "rigId": "3377974952",
            "subPoses": [
                {
                    "status": "uninitialized",
                    "pose": {
                        "rotation": [
                            "1",
                            "0",
                            "0",
                            "0",
                            "1",
                            "0",
                            "0",
                            "0",
                            "1"
                        ],
                        "center": [
                            "0",
                            "0",
                            "0"
                        ]
                    }
                },
                {
                    "status": "uninitialized",
                    "pose": {
                        "rotation": [
                            "1",
                            "0",
                            "0",
                            "0",
                            "1",
                            "0",
                            "0",
                            "0",
                            "1"
                        ],
                        "center": [
                            "0",
                            "0",
                            "0"
                        ]
                    }
                }
            ]
        }
    ]
}
```

And started it with (tried it with gui to):

C:\Users\Xma1\Desktop\Meshroom-2021.1.0\aliceVision\bin\aliceVision_rigCalibration.exe --sfmdata "C:/Users/Xma1/Desktop/RIGCHECK/cameras.sfm" --mediapath "C:/Users/Xma1/Desktop/RIGCHECK/rig2" --cameraIntrinsics "" --export "trackedcameras.abc" --descriptorPath "C:/Users/Xma1/Desktop/RIGCHECK/MeshroomCache/FeatureExtraction/54f7cb26c30018d5c72e877dd2e64093752516a7" --matchDescTypes sift --preset normal --resectionEstimator acransac --matchingEstimator acransac --refineIntrinsics "" --reprojectionError 4.0 --maxInputFrames 0 --voctree "C:\Users\Xma1\Desktop\Meshroom-2021.1.0\aliceVision\share\aliceVision\vlfeat_K80L3.SIFT.tree" --voctreeWeights "" --algorithm AllResults --nbImageMatch 4 --maxResults 10 --matchingError 4.0 --nNearestKeyFrames 5 --outfile "C:/Users/Xma1/Desktop/RIGCHECK/MeshroomCache/CameraRigCalibration/b1a2e820962e78196cda5068fe72cd3b64b0502e/cameraRigCalibration.rigCal"

C:\Users\Xma1\Desktop\Meshroom-2021.1.0\aliceVision\bin\aliceVision_rigCalibration.exe --sfmdata "C:/Users/Xma1/Desktop/RIGCHECK/MeshroomCache/StructureFromMotion/6690c080eef5b22e67b337953828c95ce98fa92b/cameras.sfm" --mediapath "C:/Users/Xma1/Desktop/RIGCHECK/rig2" --cameraIntrinsics "" --export "trackedcameras.abc" --descriptorPath "C:/Users/Xma1/Desktop/RIGCHECK/MeshroomCache/FeatureExtraction/54f7cb26c30018d5c72e877dd2e64093752516a7" --matchDescTypes sift --preset normal --resectionEstimator acransac --matchingEstimator acransac --refineIntrinsics "" --reprojectionError 4.0 --maxInputFrames 0 --voctree "C:\Users\Xma1\Desktop\Meshroom-2021.1.0\aliceVision\share\aliceVision\vlfeat_K80L3.SIFT.tree" --voctreeWeights "" --algorithm AllResults --nbImageMatch 4 --maxResults 10 --matchingError 4.0 --nNearestKeyFrames 5 --outfile "C:/Users/Xma1/Desktop/RIGCHECK/MeshroomCache/CameraRigCalibration/b1a2e820962e78196cda5068fe72cd3b64b0502e/cameraRigCalibration.rigCal"

but I get:
C:\Users\Xma1>C:\Users\Xma1\Desktop\Meshroom-2021.1.0\aliceVision\bin\aliceVision_rigCalibration.exe --sfmdata "C:/Users/Xma1/Desktop/RIGCHECK/cameras.sfm" --mediapath "C:/Users/Xma1/Desktop/RIGCHECK/rig2" --cameraIntrinsics "" --export "trackedcameras.abc" --descriptorPath "C:/Users/Xma1/Desktop/RIGCHECK/MeshroomCache/FeatureExtraction/54f7cb26c30018d5c72e877dd2e64093752516a7" --matchDescTypes sift --preset normal --resectionEstimator acransac --matchingEstimator acransac --refineIntrinsics "" --reprojectionError 4.0 --maxInputFrames 0 --voctree "C:\Users\Xma1\Desktop\Meshroom-2021.1.0\aliceVision\share\aliceVision\vlfeat_K80L3.SIFT.tree" --voctreeWeights "" --algorithm AllResults --nbImageMatch 4 --maxResults 10 --matchingError 4.0 --nNearestKeyFrames 5 --outfile "C:/Users/Xma1/Desktop/RIGCHECK/MeshroomCache/CameraRigCalibration/b1a2e820962e78196cda5068fe72cd3b64b0502e/cameraRigCalibration.rigCal"
Program called with the following parameters:
 * algorithm = "AllResults"
 * cameraIntrinsics =  = []
 * descriptorPath = "C:/Users/Xma1/Desktop/RIGCHECK/MeshroomCache/FeatureExtraction/54f7cb26c30018d5c72e877dd2e64093752516a7"
 * export = "trackedcameras.abc"
 * matchDescTypes = "sift"
 * matchingError = 4
 * matchingEstimator =  Unknown Type "enum aliceVision::robustEstimation::ERobustEstimator"
 * maxInputFrames = 0
 * maxResults = 10
 * mediapath =  = [C:/Users/Xma1/Desktop/RIGCHECK/rig2]
 * nNearestKeyFrames = 5
 * nbImageMatch = 4
 * outfile = "C:/Users/Xma1/Desktop/RIGCHECK/MeshroomCache/CameraRigCalibration/b1a2e820962e78196cda5068fe72cd3b64b0502e/cameraRigCalibration.rigCal"
 * preset =  Unknown Type "enum aliceVision::feature::EImageDescriberPreset"
 * randomSeed = 5489 (default)
 * refineIntrinsics = 1
 * reprojectionError = 4
 * resectionEstimator =  Unknown Type "enum aliceVision::robustEstimation::ERobustEstimator"
 * sfmdata = "C:/Users/Xma1/Desktop/RIGCHECK/cameras.sfm"
 * voctree = "C:\Users\Xma1\Desktop\Meshroom-2021.1.0\aliceVision\share\aliceVision\vlfeat_K80L3.SIFT.tree"
 * voctreeWeights = ""

Calibrating sequence using the voctree localizer
[2021-04-16 15:02:14.848176] [0x00002984] [error]   cudaGetDeviceCount failed: CUDA driver version is insufficient for CUDA runtime version
[2021-04-16 15:02:14.849151] [0x00002984] [info]    Can't find CUDA-Enabled GPU.
[2021-04-16 15:02:14.850129] [0x00002984] [debug]   Loading vocabulary tree...
[2021-04-16 15:02:15.049641] [0x00002984] [debug]   tree loaded with 3 levels and 80 branching factors
[2021-04-16 15:02:15.049641] [0x00002984] [debug]   Creating the database...
[2021-04-16 15:02:15.058429] [0x00002984] [debug]   No weights specified, skipping...
[2021-04-16 15:02:15.058429] [0x00002984] [debug]   Build observations per view

- Load Features and Descriptors per view -
0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
CAMERA 0
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

[2021-04-16 15:02:15.071640] [0x00002984] [debug]   folder without expression: C:/Users/Xma1/Desktop/RIGCHECK/rig2
[2021-04-16 15:02:15.076006] [0x00002984] [debug]   directory feedImage
Input stream length is 0.

******************************
Processed 0 images for camera 0
Processing took 0 [s] overall
Mean time for localization:   -nan(ind) [ms]
Max time for localization:   -1.79769e+308 [ms]
Min time for localization:   1.79769e+308 [ms]

******************************
Camera 0 localized 0/0
Rig calibration initialization...
[2021-04-16 15:02:15.089674] [0x00002984] [debug]   The calibration results are empty!
Unable to find a proper initialization for the relative poses! Aborting...

C:\Users\Xma1\Desktop\Meshroom-2021.1.0\aliceVision\bin\aliceVision_rigCalibration.exe --sfmdata "C:/Users/Xma1/Desktop/RIGCHECK/cameras.sfm" --mediapath "C:/Users/Xma1/Desktop/RIGCHECK/rig2" --cameraIntrinsics "" --export "trackedcameras.abc" --descriptorPath "C:/Users/Xma1/Desktop/RIGCHECK/MeshroomCache/FeatureExtraction/54f7cb26c30018d5c72e877dd2e64093752516a7" --matchDescTypes sift --preset normal --resectionEstimator acransac --matchingEstimator acransac --refineIntrinsics "" --reprojectionError 4.0 --maxInputFrames 0 --voctree "C:\Users\Xma1\Desktop\Meshroom-2021.1.0\aliceVision\share\aliceVision\vlfeat_K80L3.SIFT.tree" --voctreeWeights "" --algorithm AllResults --nbImageMatch 4 --maxResults 10 --matchingError 4.0 --nNearestKeyFrames 5 --outfile "C:/Users/Xma1/Desktop/RIGCHECK/MeshroomCache/CameraRigCalibration/b1a2e820962e78196cda5068fe72cd3b64b0502e/cameraRigCalibration.rigCal"
Program called with the following parameters:
 * algorithm = "AllResults"
 * cameraIntrinsics =  = []
 * descriptorPath = "C:/Users/Xma1/Desktop/RIGCHECK/MeshroomCache/FeatureExtraction/54f7cb26c30018d5c72e877dd2e64093752516a7"
 * export = "trackedcameras.abc"
 * matchDescTypes = "sift"
 * matchingError = 4
 * matchingEstimator =  Unknown Type "enum aliceVision::robustEstimation::ERobustEstimator"
 * maxInputFrames = 0
 * maxResults = 10
 * mediapath =  = [C:/Users/Xma1/Desktop/RIGCHECK/rig2]
 * nNearestKeyFrames = 5
 * nbImageMatch = 4
 * outfile = "C:/Users/Xma1/Desktop/RIGCHECK/MeshroomCache/CameraRigCalibration/b1a2e820962e78196cda5068fe72cd3b64b0502e/cameraRigCalibration.rigCal"
 * preset =  Unknown Type "enum aliceVision::feature::EImageDescriberPreset"
 * randomSeed = 5489 (default)
 * refineIntrinsics = 1
 * reprojectionError = 4
 * resectionEstimator =  Unknown Type "enum aliceVision::robustEstimation::ERobustEstimator"
 * sfmdata = "C:/Users/Xma1/Desktop/RIGCHECK/cameras.sfm"
 * voctree = "C:\Users\Xma1\Desktop\Meshroom-2021.1.0\aliceVision\share\aliceVision\vlfeat_K80L3.SIFT.tree"
 * voctreeWeights = ""

Calibrating sequence using the voctree localizer
[2021-04-16 16:35:46.219720] [0x0000620c] [error]   cudaGetDeviceCount failed: CUDA driver version is insufficient for CUDA runtime version
[2021-04-16 16:35:46.220695] [0x0000620c] [info]    Can't find CUDA-Enabled GPU.
[2021-04-16 16:35:46.221671] [0x0000620c] [debug]   Loading vocabulary tree...
[2021-04-16 16:35:46.382766] [0x0000620c] [debug]   tree loaded with 3 levels and 80 branching factors
[2021-04-16 16:35:46.382766] [0x0000620c] [debug]   Creating the database...
[2021-04-16 16:35:46.391552] [0x0000620c] [debug]   No weights specified, skipping...
[2021-04-16 16:35:46.391552] [0x0000620c] [debug]   Build observations per view

- Load Features and Descriptors per view -
0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
CAMERA 0
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

[2021-04-16 16:35:46.398371] [0x0000620c] [debug]   folder without expression: C:/Users/Xma1/Desktop/RIGCHECK/rig2
[2021-04-16 16:35:46.398371] [0x0000620c] [debug]   directory feedImage
Input stream length is 0.

******************************
Processed 0 images for camera 0
Processing took 0 [s] overall
Mean time for localization:   -nan(ind) [ms]
Max time for localization:   -1.79769e+308 [ms]
Min time for localization:   1.79769e+308 [ms]

******************************
Camera 0 localized 0/0
Rig calibration initialization...
[2021-04-16 16:35:46.410089] [0x0000620c] [debug]   The calibration results are empty!
Unable to find a proper initialization for the relative poses! Aborting...

**Comment by natowi** [comment]
> I have no exif in images

> Requirements:
The EXIF serial number (BodySerialNumber) for each image set are required to be different. (Images with the same EXIF information are ignored) Cameras need to move/capture in sync like a stereo rig. In case of a turntable, cameras should capture in sync to the rotation of the turntable from their fixed positions.
https://github.com/alicevision/meshroom/wiki/Multi-Camera-Rig

At the moment the Rig support is highly sensitive to configuration / metadata issues.

So it is possible the rig is not defined properly due to the missing metadata.

**Reply by markxma1** [reply]
thank for answer, but injecting metadata and converting into jpg haven't worked. 

I have added:
```
FocalLength                     : 16.0 mm
SerialNumber                    : 101 //for camera 1 it is 101,102... and camera 2 is 201, 202...
Model                           : Canon 550D //my camera is not in the list and I don't have full Information now, I Have only images.
Make                            : Canon
FocalLength35efl                : 16.0 mm
```

do I need to add extra camera Intrinsics? 
If yes, how can i generate it from sfmdata.
I think it is saved into sfm data. 

```

C:\Users\xma1\Desktop\rig\Meshroom-2021.1.0\aliceVision\bin>aliceVision_rigCalibration  --sfmdata "C:/Users/xma1/Desktop/rig/RIGCHECK/MeshroomCache/CameraInit/d744ebd2326a83286bd658a29bed3eac0b603f8f/cameraInit.sfm" --mediapath "C:/Users/xma1/Desktop/rig/RIGCHECK/rig" --cameraIntrinsics "" --export "trackedcameras.abc" --descriptorPath "C:/Users/xma1/Desktop/rig/RIGCHECK/MeshroomCache/FeatureExtraction/084e35aa36084ec099f18824003f89c34c99854d" --matchDescTypes sift --preset normal --resectionEstimator acransac --matchingEstimator acransac --refineIntrinsics "" --reprojectionError 4.0 --maxInputFrames 0 --voctree "C:\Users\xma1\Desktop\rig\Meshroom-2021.1.0\aliceVision\share\aliceVision\vlfeat_K80L3.SIFT.tree" --voctreeWeights "" --algorithm AllResults --nbImageMatch 4 --maxResults 10 --matchingError 4.0 --nNearestKeyFrames 5 --outfile "C:/Users/xma1/Desktop/rig/RIGCHECK/MeshroomCache/CameraRigCalibration/38fff4349111fbd3905c1baf8891ef0c8b8f75e9/cameraRigCalibration.rigCal"
Program called with the following parameters:
 * algorithm = "AllResults"
 * cameraIntrinsics =  = []
 * descriptorPath = "C:/Users/xma1/Desktop/rig/RIGCHECK/MeshroomCache/FeatureExtraction/084e35aa36084ec099f18824003f89c34c99854d"
 * export = "trackedcameras.abc"
 * matchDescTypes = "sift"
 * matchingError = 4
 * matchingEstimator =  Unknown Type "enum aliceVision::robustEstimation::ERobustEstimator"
 * maxInputFrames = 0
 * maxResults = 10
 * mediapath =  = [C:/Users/xma1/Desktop/rig/RIGCHECK/rig]
 * nNearestKeyFrames = 5
 * nbImageMatch = 4
 * outfile = "C:/Users/xma1/Desktop/rig/RIGCHECK/MeshroomCache/CameraRigCalibration/38fff4349111fbd3905c1baf8891ef0c8b8f75e9/cameraRigCalibration.rigCal"
 * preset =  Unknown Type "enum aliceVision::feature::EImageDescriberPreset"
 * randomSeed = 5489 (default)
 * refineIntrinsics = 1
 * reprojectionError = 4
 * resectionEstimator =  Unknown Type "enum aliceVision::robustEstimation::ERobustEstimator"
 * sfmdata = "C:/Users/xma1/Desktop/rig/RIGCHECK/MeshroomCache/CameraInit/d744ebd2326a83286bd658a29bed3eac0b603f8f/cameraInit.sfm"
 * voctree = "C:\Users\xma1\Desktop\rig\Meshroom-2021.1.0\aliceVision\share\aliceVision\vlfeat_K80L3.SIFT.tree"
 * voctreeWeights = ""

[2021-04-16 23:40:04.491567] [0x00006108] [trace]   0 extrinsics declared and 8 extrinsics used.
Calibrating sequence using the voctree localizer
[2021-04-16 23:40:04.551910] [0x00006108] [info]    Supported CUDA-Enabled GPU detected.
[2021-04-16 23:40:04.551910] [0x00006108] [debug]   Loading vocabulary tree...
[2021-04-16 23:40:04.641992] [0x00006108] [debug]   tree loaded with 3 levels and 80 branching factors
[2021-04-16 23:40:04.641992] [0x00006108] [debug]   Creating the database...
[2021-04-16 23:40:04.645995] [0x00006108] [debug]   No weights specified, skipping...
[2021-04-16 23:40:04.645995] [0x00006108] [debug]   Build observations per view

- Load Features and Descriptors per view -
0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
CAMERA 0
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


[2021-04-16 23:40:04.652001] [0x00006108] [debug]   folder without expression: C:/Users/xma1/Desktop/rig/RIGCHECK/rig
[2021-04-16 23:40:04.652001] [0x00006108] [debug]   directory feedImage
Input stream length is 0.


******************************
Processed 0 images for camera 0
Processing took 0 [s] overall
Mean time for localization:   -nan(ind) [ms]
Max time for localization:   -1.79769e+308 [ms]
Min time for localization:   1.79769e+308 [ms]


******************************
Camera 0 localized 0/0
Rig calibration initialization...
[2021-04-16 23:40:04.663011] [0x00006108] [debug]   The calibration results are empty!
Unable to find a proper initialization for the relative poses! Aborting...
```

**Comment by natowi** [comment]
> Model                           : Canon 550D //my camera is not in the list and I don't have full Information now, I Have only images.

Your camera is listed in the sensordb:
Canon;Canon EOS 550D;22.3;dpreview,digicamdb,imaging-resource,dxomark
It should match with Canon 550D, too. Does the green icon show up in the image viewer when importing the image?
Make sure your metadata is being detected correctly. (Adding Make and focallength as you did is good, too.)


> SerialNumber                    : 101 //for camera 1 it is 101,102... and camera 2 is 201, 202...

No, we don´t need a serial number for each image, but for each camera:

+ rig/  # "rig" folder
|-+ 0/  # sub-folder with the index of the camera (starting at 0)
|---- 101.png - - serialnr: 1 - - timestamp: 00:01
|---- 102.png - - serialnr: 1 - - timestamp: 00:02
|-+ 1/ # sub-folder with the index of the camera
|---- 101.png - - serialnr: 2 - - timestamp: 00:01
|---- 102.png- - serialnr: 2 - - timestamp: 00:02

(Side note: having the same serial number for 101.png from camera 1 and 101.png from camera 2 would result in only one image being imported, as Meshroom would assume it to be the same image, since the timestamp and filename is identical. Error message at the end of the page -> https://github.com/alicevision/meshroom/wiki/Multi-Camera-Rig)

When you drop the rig sub-folders into Meshroom, the rig constraint will be recognized and you will be able to see it in the CameraInit node (see Rig and Rig Sub-Pose of the Viewpoints parameter). Check if that is the case.
Also see if the cli window throws any errors.

**Reply by markxma1** [reply]
I will try to add timestamp to it, maby it will help somehow. 

Sorry for confusing it with the camera, I chose Canon because my camera (from Basler) is not in the list and I don't know the camera name, just the manufacturer. I'm now not near to the camera and cannot check parameter and name of it. Maybe I'll build small construction with ESP32 that I can take with me everywhere.

I tried both options by (1) giving each camera the serial number and (2) each picture the serial number. Images have been loaded into Meshroom, that's no problem. It runs to StructurFromMotion (Dense and other steps was removed), but the cameras are not tied to a rig, and when I use  CameraRigCalibration, it throws an error. 
I loaded cameras into Blender and inserted camera 2 in camera 1 (reparent with keep transformation) and set each camera 1 to position and rotation 0, camera 2 has a slightly different position, which normally cannot be in the rig, cameras bounded tied to aluminum frame and can't move anyway (ok, with great force, but not in this case).

names are normaly 1-9 (with leading zeros if more digits). „10x“ is only to separate names in other folders (cameraID0ImageID) without use serial number or to code camera and imageid into serial number.

rig/ # "rig" folder
|-+ 0/ # sub-folder with the index of the camera (starting at 0)
|---- 1.png - - serialnr: 1 
|---- 2.png - - serialnr: 1 
|-+ 1/ # sub-folder with the index of the camera
|---- 1.png - - serialnr: 2 
|---- 2.png- - serialnr: 2

If i use 101 and 201 names i can see rig and rig sub-pose to (and i can change to matching rig id) and in Structure from Motion is some checkbox with rig. I had tried it before I ascend in the forum, but I don't see any difference whether it is with the rig checkbox or without in this case. The cameras are not connected and have different positions to each other.

I can not start CameraRigCalibration from Utils/CameraRigCalibration. 

Maybe I misunderstood something and the camera calibration is not there to improve the camera position in the camera rig housing (so that one camera improves the position of another camera to avoid a local minimum. So that the cameras have the same constant position to each other and only move in groups.)

**Comment by natowi** [comment]
Please try the default graph first and make sure it works before adding other nodes. CameraInit should detect the rig correctly and the SfM should compute without errors. You could share a portion of the (raw, unmodified) dataset if you want and I can take a look. 


CameraRigCalibration is optional - details: https://meshroom-manual.readthedocs.io/en/latest/feature-documentation/nodes/CameraRigCalibration.html

**Comment by markxma1** [comment]
here are images and Blender file where i put cameras to first camera and moved first camera to Zero. After Metadata changes much better, but there is still difference. 

If i fill one camera with black color (to emit that camera have no points), camera is removed (on rig we can get position of that camera with help of another).

https://www.dropbox.com/t/SBV7ZmZf8FE9Onci

and where is example for /path/to/calib1.txt for CameraRigCalibration to get relative poses between cameras of the rig and choose the more stable value across the images?

**Comment by natowi** [comment, marked as answer]
I´ll take a look.


> and where is example for /path/to/calib1.txt for CameraRigCalibration to get relative poses between cameras of the rig and choose the more stable value across the images?

You can calibrate each camera with https://meshroom-manual.readthedocs.io/en/latest/feature-documentation/nodes/CameraCalibration.html
As you don´t have access to your camera you can not use camera calibration, as it requires images/videos of a calibration pattern.


This is how the correctly set up rig should look like in camerainit. If rig pose and/or sub pose is -1 something went wrong for two cameras there should also be only two intrinsics:

(ignore the 2019.1 name in the path, it´s just where I put the image files)
![Unbenannt](https://user-images.githubusercontent.com/8138537/115126125-2f8fd700-9fcd-11eb-956c-366525fb1faa.JPG)

![2021-04-17_22-30-26](https://user-images.githubusercontent.com/8138537/115126130-34548b00-9fcd-11eb-8600-1fceec97e4ac.jpg)

Your dataset has 8 intrinsics declared, only two are expected:

![new](https://user-images.githubusercontent.com/8138537/115126277-22bfb300-9fce-11eb-91d8-5b303e08ca26.JPG)

Checking the metadata I noticed your PixelXDimension is changing between images! Images should always have the same,fixed dimensions.

![new2](https://user-images.githubusercontent.com/8138537/115126388-e3459680-9fce-11eb-95ae-554182910bc0.JPG)

The BodySerialNumber should be different for each camera but stay the same for images taken from the same camera. each of the two cameras chould have a unique serialnumber. The changing PixelXDimension results in Meshroom assuming you have 8 different cameras, as changing the x dimension is not expected for the same camera. Never modify (crop/distort) images prior import for photogrammetry use.

**Reply by markxma1** [reply]
thanks for the reply i did not notice that the size changes with the standard parameters of the camera. I don't crop or resize image.  I will set the capture script to the manual size parameter next week. And sorry for BodySerialNumber, I tried 1 for camera 1 and 2 for camera 2 first and then (because it doesn't worked) changed it to 101 and forgot to change it again.

**Reply by natowi** [reply]
> And sorry for BodySerialNumber, I tried 1 for camera 1 and 2 for camera 2 first and then (because it doesn't worked) changed it to 101 and forgot to change it again.

it´s fine 1 and 2 were only placeholders. With the dataset you provided it changed within a camera dataset where it is expected to be the same for all files

**Reply by markxma1** [reply]
orks, I need to set a rig ID for all rigs (I thought it is for each position). Now everything is combined into one rig. Many thanks for the help.
